packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts

```diff
@@ -94,8 +93,42 @@ describe('Portfolio', () => {
           description: description,
         });
       }).toThrowError(/Invalid portfolio description for resource Default\/MyPortfolio/);
+    }),
+
+    test('portfolio creation with token description does not throw validation error', () => {
+      const tokenDescription = cdk.Lazy.string({ produce: () => 'token invalid description'.repeat(1000) });
+
+      expect(() => {
+        new servicecatalog.Portfolio(stack, 'MyPortfolio', {
+          displayName: 'testPortfolio',
+          providerName: 'testProvider',
+          description: tokenDescription,
+        });
+      }).not.toThrowError(/Invalid portfolio description for resource Default\/MyPortfolio/);
+    }),
+
+    test('portfolio creation with token display name does not throw validation error', () => {
+      const tokenDisplayName = cdk.Lazy.string({ produce: () => 'token invalid displayName'.repeat(1000) });
+
+      expect(() => {
+        new servicecatalog.Portfolio(stack, 'MyPortfolio', {
+          displayName: tokenDisplayName,
+          providerName: 'testProvider',
+        });
+      }).not.toThrowError(/Invalid portfolio display name for resource Default\/MyPortfolio/);
+    }),
+
+    test('portfolio creation with token provider name does not throw validation error', () => {
+      const tokenProviderName = cdk.Lazy.string({ produce: () => '' });
+
+      expect(() => {
+        new servicecatalog.Portfolio(stack, 'MyPortfolio', {
+          displayName: 'testPortfolio',
+          providerName: tokenProviderName,
+        });
+      }).not.toThrowError(/Invalid portfolio provider name for resource Default\/MyPortfolio/);
```

Comment:
Same comment here as in https://github.com/aws/aws-cdk/pull/15140#discussion_r654716111 for all these new tests:

1. Use `CfnParameter`, not `Lazyy`.
2. Assert success by looking at the template, not `not.toThrow()` (this doesn't exercise the regex passed to `thThrowError()`, which means it can get out of date).

Reply:
Looking through some other examples of this and did not really see design that I liked.  Would this be what you think is best?
```
    test('portfolio creation with token description does not throw validation error', () => {
      const tokenDescription = new cdk.CfnParameter(stack, 'Description');

      new servicecatalog.Portfolio(stack, 'MyPortfolio', {
        displayName: 'testPortfolio',
        providerName: 'testProvider',
        description: tokenDescription.valueAsString,
      });

      expect(stack).toHaveResourceLike('AWS::ServiceCatalog::Portfolio', {
        Description: {
          Ref: 'Description',
        },
      });
    });
```
Or would it be better to pass in a `descriptionString` to the `CfnParameter` and check `Ref: descriptionString` ?

Reply:
The code you have there is exactly how I would test it.